plugins/chat/plugin.rb

```diff
@@ -9,253 +9,36 @@
-register_asset "stylesheets/common/chat-composer-inline-button.scss"
-register_asset "stylesheets/common/chat-replying-indicator.scss"
-register_asset "stylesheets/common/chat-composer.scss"
-register_asset "stylesheets/desktop/chat-composer.scss", :desktop
-register_asset "stylesheets/mobile/chat-composer.scss", :mobile
-register_asset "stylesheets/common/direct-message-creator.scss"
-register_asset "stylesheets/common/chat-message-collapser.scss"
-register_asset "stylesheets/common/chat-message-images.scss"
-register_asset "stylesheets/common/chat-transcript.scss"
-register_asset "stylesheets/common/chat-composer-dropdown.scss"
-register_asset "stylesheets/common/chat-retention-reminder.scss"
-register_asset "stylesheets/common/chat-composer-uploads.scss"
-register_asset "stylesheets/desktop/chat-composer-uploads.scss", :desktop
-register_asset "stylesheets/common/chat-composer-upload.scss"
-register_asset "stylesheets/common/chat-selection-manager.scss"
-register_asset "stylesheets/mobile/chat-selection-manager.scss", :mobile
-register_asset "stylesheets/common/chat-channel-selector-modal.scss"
-register_asset "stylesheets/mobile/mobile.scss", :mobile
-register_asset "stylesheets/desktop/desktop.scss", :desktop
-register_asset "stylesheets/sidebar-extensions.scss"
-register_asset "stylesheets/desktop/sidebar-extensions.scss", :desktop
-register_asset "stylesheets/common/chat-message-actions.scss"
-register_asset "stylesheets/desktop/chat-message-actions.scss", :desktop
-register_asset "stylesheets/mobile/chat-message-actions.scss", :mobile
-register_asset "stylesheets/common/chat-message-separator.scss"
-register_asset "stylesheets/common/chat-onebox.scss"
-register_asset "stylesheets/common/chat-skeleton.scss"
 register_asset "stylesheets/colors.scss", :color_definitions
-register_asset "stylesheets/common/reviewable-chat-message.scss"
-register_asset "stylesheets/common/chat-mention-warnings.scss"
-register_asset "stylesheets/common/chat-channel-settings-saved-indicator.scss"
-register_asset "stylesheets/common/chat-thread.scss"
-register_asset "stylesheets/common/chat-side-panel.scss"
+register_asset "stylesheets/mixins/index.scss"
+register_asset "stylesheets/common/index.scss"
+register_asset "stylesheets/desktop/index.scss", :desktop
+register_asset "stylesheets/mobile/index.scss", :mobile
 
 register_svg_icon "comments"
 register_svg_icon "comment-slash"
 register_svg_icon "hashtag"
 register_svg_icon "lock"
-
 register_svg_icon "file-audio"
 register_svg_icon "file-video"
 register_svg_icon "file-image"
 
 # route: /admin/plugins/chat
 add_admin_route "chat.admin.title", "chat"
 
-# Site setting validators must be loaded before initialize
-require_relative "lib/validators/chat_default_channel_validator.rb"
-require_relative "lib/validators/chat_allow_uploads_validator.rb"
-require_relative "lib/validators/direct_message_enabled_groups_validator.rb"
 require_relative "app/core_ext/plugin_instance.rb"
```

Comment:
Now that `app/**/*` is autoloaded, this file is automatically loaded when Zeitwerk is in `eager_load` mode (i.e. in production). Since this file does not define a `PluginInstance` constant, it raises an error:

```
/Users/david/.rvm/gems/ruby-3.2.1/gems/zeitwerk-2.6.7/lib/zeitwerk/loader/helpers.rb:135:in `const_get': uninitialized constant PluginInstance (NameError)

    parent.const_get(cname, false)
          ^^^^^^^^^^
Did you mean?  PluginStore
```

At a minimum, we need to move it outside of `app/...`. Even better would be to follow the same `reloadable_patch { Blah.prepend Chat::BlahExtension }` pattern we use below.